resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/LoggingPodStatusWatcher.scala

```diff
@@ -177,4 +188,27 @@ private[k8s] class LoggingPodStatusWatcherImpl(
   private def formatTime(time: String): String = {
     if (time != null ||  time != "") time else "N/A"
   }
+
+  override def watchOrStop(sId: String): Boolean = if (hasCompleted()) {
```

Comment:
This line is different in the `master` branch. It check whether the client was set to wait for the completion of the driver, not simply checking `hasCompleted()`, which should normally return false - causing `spark-submit` to terminate immediately. 

For `2.4.7` port you may want to add additional parameter to the `LoggingPodStatusWatcherImpl`, specifically `waitForCompletion: Boolean`, that can be checked in this if statement. In current master they are using `KubernetesDriverConf` instead of passing parameters separately: https://github.com/apache/spark/blob/5e6173ebefcbf3d85ba10234b35fe043f8497d32/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/LoggingPodStatusWatcher.scala#L40. 

However in `2.4.6`, `KubernetesDriverConf` is not yet introduced, and instead the `appId` and `maybeLoggingInterval` are passed individually. Simple solution is to add `waitForCompletion` as a parameter, and use it in the above if statement.

Reply:
@shockdm  Yes, you are right.  And it is great that you have it working!
We're happy to make the change but would you mind sharing the code so as to be consistent with what you already have?

Reply:
@shockdm ping

Reply:
@jkleckner sorry, spent some time doing test runs to make sure it fixed our use case. I'll share the final code shortly.

Reply:
@shockdm ping

Reply:
@jkleckner Thank you for the reminder, and sorry for the delay! Check out https://github.com/shockdm/spark/pull/1

Reply:
Your patches look like an improvement, thank you @shockdm.
There are enough differences that perhaps you might be willing to submit them?

Reply:
With the original comment by @dongjoon-hyun:

> Thank you for making a backporting PR, @jkleckner .
Could you make a backporting PR to branch-3.0 first?
To prevent any regression from 2.4 to 3.0, we have a policy for backporting sequence. We cannot backport this to branch-2.4 directly.

What would be the correct approach to submit this backport atm?

Reply:
@dongjoon-hyun Is the merging or the submitting of a PR to branch-3.0 a pre-requisite to submitting this PR or an improved PR?

Note that the implementations inherently diverge due to the divergence of the base branches.

Reply:
@shockdm The 3.0 patch #29533 has now been merged.  Even though we never got an answer whether that was a prerequisite, that blockage is now gone.

Would you be willing to rebase and clean up your patch for submission?